### PR TITLE
[user-managerd] Fix user id range and avoid running removeUserFiles for system users. JB#58459

### PR DESCRIPTION
**Pull request by krnlyng**

### CODE REVIEW

File: src/sailfishusermanager.cpp
Comment (Tomin1) [inline src/sailfishusermanager.cpp]: I would maybe define a new variable for the maximum UID instead of `OWNER_USER_ID + MAX_RESERVED_UID`. Maybe `MAX_USER_UID` or something, feel free to suggest.

Additionally maybe this could be on a single line with the previous `if`:
```
if (pwd && pwd->pw_uid >= OWNER_USER_UID && pwd->pw_uid <= MAX_USER_UID) {
```

Comment (Tomin1) [reply src/sailfishusermanager.cpp]: And then could also add a static assert to the beginning that `SAILFISH_USERMANAGER_GUEST_UID` is within this limit.

Comment (Tomin1) [reply src/sailfishusermanager.cpp]: I guess technically the strictest we could limit this to is `[OWNER_USER_UID, OWNER_USER_UID + SAILFISH_USERMANAGER_MAX_USERS[ | SAILFISH_USERMANAGER_GUEST_UID`, but it's debatable what's better. :smiley:

**Review by Tomin1 — APPROVED** [review]
LGTM now, let's check if Simo has some comment or not.

And just to clarify: It's also fine by me to merge this without his approval. He might not have time to check this.

**Review by spiiroin — APPROVED** [review]

**Review by mkosola — APPROVED** [review]

**Pull request merged**
